source/codegen/metadata/nidcpower/enums.py

```diff
@@ -144,6 +144,18 @@
             }
         ]
     },
+    'PowerLineFrequencies': {
```

Comment:
This constant is of type Real64 and is used in a Real64 attribute and Real64 parameter in API. but it has been added as Int32 values because enum only supports that datatype and the values of the enum (50, 60) are integers

Reply:
fixed

Reply:
Jonathan's out today and half of Wednesday and Thursday. But from my perspective, this seems alright for now since like you said we only currently support Int32 enum values.

If PowerLineFrequencies values update in the future to contain non-integer values we may have to reconsider if we want to add support in codegen for non-int32 enums. I believe we had started to code what that would like for a few enums in ni-fake but didn't complete it to be used for actual APIs.

Reply:
In `nidcpower.h` these constants are still defined as `50.0` and `60.0`. I am not C expert, but off the top your head do you think it could cause some issues in dcpower's C library when comparing/doing any operations with user passed `50` with constant in header `50.0`?

In DMM we do have some enums with float values as `0.0001` etc. We were thinking of just documenting these possible values for parameter in wiki and not define enums for these in gRPC. I think we have done same for some string enums in Switch. @reckenro do you think this approach is fine? What does it take to add non-in32 enum support in codegen?

Reply:
I am no C expert either, but I believe that should be fine to my knowledge.

I think I talked briefly to Akshata about that and yes, if we don't change anything I think that's the route to take. Document those possible values and disregard the enum in enums.py and references in the parameters so the client has to provide the actual raw value. I think that's fine and that's what we decided on initially to just support int32 enums.

I believe we have partial support for non-int32 enum support in the codegen currently. We had begun to go down this route but abandoned it when we decided to support int32 enums only but the code remained. If you tag an enum in enums.py with `'generate-mappings': True` the codegen should try to generate code such that the proto file has {0, 1, 2, ...} values for the enum and the service implementation has mappings stored in the header file to map those valid enum values to the actual double or string values proto_enum_to_value_mapping = {{0, 0.1} {1, 0.01}, {2, 0.02}}.
The service implementation methods will then know to use this dictionary if it comes a cross an enum parameter that has the `'generate-mappings'` tag.

That's the basic idea and if you look in ni-fake's enums.py I believe we have a few enums with that tag that are used in a function so the generated ni-fake service is generating in this way. That being said, I think there's remaining work to validate that it works as expected and make sure there aren't any gaps in its logic since we never adopted its usage in an actual driver API.